core/src/main/scala/org/apache/spark/SecurityManager.scala

```diff
@@ -310,7 +310,13 @@ private[spark] class SecurityManager(sparkConf: SparkConf)
     setViewAcls(Set[String](defaultUser), allowedUsers)
   }
 
-  def getViewAcls: String = viewAcls.mkString(",")
+  def getViewAcls: String = {
+    if (viewAcls.contains("*")) {
```

Comment:
It would be good here to put a comment that YARN requires only a * to be returned.   you can't have *,user1,user2 to explain why we do this